client-spark/common/src/main/java/org/apache/spark/shuffle/RssShuffleUtils.java

```diff
@@ -119,7 +119,7 @@
   public static ByteBuffer decompressData(ByteBuffer data, int uncompressLength) {
     LZ4FastDecompressor fastDecompressor = LZ4Factory.fastestInstance().fastDecompressor();
     ByteBuffer uncompressData = ByteBuffer.allocateDirect(uncompressLength);
-    fastDecompressor.decompress(data, 0, uncompressData, 0, uncompressLength);
+    fastDecompressor.decompress(data, data.position(), uncompressData, 0, uncompressLength);
```

Comment:
why do we use `data.postition`?

Reply:
position means the start point, it's the right value here. The 0 is a bug, it works because the start point is always 0 before this PR.

Reply:
What's the influence of this bug? Should we add some tests?

Reply:
with previous implementation, if data.position != 0, the decompress will failed.
The current UT can catch the bug, it's ok without new UT.

Reply:
Spark application will fail sometimes？